server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -453,6 +460,24 @@ public void onFailure(Exception e) {
                     continue;
                 }
 
+                // Check if index closing conflicts with any running restores
+                Set<Index> restoringIndices = RestoreService.restoringIndices(currentState, index::equals);
+                if (restoringIndices.isEmpty() == false) {
+                    closingResults.put(result.getKey(), new IndexResult(result.getKey(), new IllegalStateException(
+                        "verification of shards before closing " + index + " succeeded but index is being restored in the meantime")));
```

Comment:
NIT: I think maybe put `[]` around the index like we generally do in exception messages?

Reply:
`Index.toString()̀` already generates a string that looks like `[name/uuid]` so we are good